src/net/sourceforge/kolmafia/KoLCharacter.java

```diff
@@ -3905,47 +3905,25 @@ private static void addAvailableSkill(final UseSkillRequest skill, final boolean
 
   /** Adds a single skill to the list of skills temporarily possessed by this character. */
   public static final void addAvailableCombatSkill(final int skillId) {
-    KoLCharacter.addAvailableCombatSkill(UseSkillRequest.getUnmodifiedInstance(skillId));
+    if (!KoLConstants.availableCombatSkillsSet.contains(skillId)) {
```

Comment:
similarly to the remove comment, all should work if you just call `add` here without contains: if it's already present, it won't be added again (as the container is a set), and if it isn't it will be.

Reply:
Done.